src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c

```diff
@@ -21,15 +33,102 @@ const ActorInit En_Dy_Extra_InitVars = {
     (ActorFunc)EnDyExtra_Update,
     (ActorFunc)EnDyExtra_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dy_Extra/EnDyExtra_Destroy.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dy_Extra/EnDyExtra_Init.s")
+void EnDyExtra_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void EnDyExtra_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnDyExtra* this = THIS;
+
+    osSyncPrintf("\n\n");
+    //"Big fairy effect"
+    osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ 大妖精効果 ☆☆☆☆☆ %d\n" VT_RST, this->actor.params);
+    this->unk_150 = this->actor.params;
+    this->unk_15C.x = 0.025f;
+    this->unk_15C.y = 0.039f;
+    this->unk_15C.z = 0.025f;
+    this->unk_168 = this->actor.posRot.pos;
+    this->actor.gravity = -0.2f;
+    this->unk_158 = 1.0f;
+    this->unk_154 = 0x3C;
+    this->actionFunc = func_809FF7AC;
+}
+
+void func_809FF7AC(EnDyExtra* this, GlobalContext* globalCtx) {
+    Math_SmoothScaleMaxF(&this->actor.gravity, 0.0f, 0.1f, 0.005f);
+    if (this->actor.posRot.pos.y < -55.0f) {
+        this->actor.velocity.y = 0.0f;
+    }
+    if (this->unk_154 == 0) {
+        if (this->unk_152 != 0) {
+            this->unk_154 = 0xC8;
+            this->actionFunc = func_809FF840;
+        }
+    }
+}
+
+void func_809FF840(EnDyExtra* this, GlobalContext* globalCtx) {
+    Math_SmoothScaleMaxF(&this->actor.gravity, 0.0f, 0.1f, 0.005f);
+    if (this->unk_154 == 0 || this->unk_158 < 0.02f) {
+        Actor_Kill(&this->actor);
+    } else {
+        Math_SmoothDownscaleMaxF(&this->unk_158, 0.03f, 0.05f);
+        if (this->actor.posRot.pos.y < -55.0f) {
+            this->actor.velocity.y = 0.0f;
+        }
+    }
+}
+
+void EnDyExtra_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnDyExtra* this = THIS;
+
+    DECR(this->unk_154);
+    this->actor.scale.x = this->unk_15C.x;
+    this->actor.scale.y = this->unk_15C.y;
+    this->actor.scale.z = this->unk_15C.z;
+    Audio_PlayActorSound2(&this->actor, NA_SE_PL_SPIRAL_HEAL_BEAM - SFX_FLAG);
+    this->actionFunc(this, globalCtx);
+    Actor_MoveForward(&this->actor);
+}
+
+void EnDyExtra_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    static Color_RGBA8_n D_809FFC40[] = { { 255, 255, 170, 255 }, { 255, 255, 170, 255 } };
+    static Color_RGBA8_n D_809FFC48[] = { { 255, 100, 255, 255 }, { 100, 255, 255, 255 } };
+    static u8 D_809FFC50[] = { 0x02, 0x01, 0x01, 0x02, 0x00, 0x00, 0x02, 0x01, 0x00, 0x02, 0x01, 0x00, 0x02,
+                               0x01, 0x00, 0x02, 0x01, 0x00, 0x02, 0x01, 0x00, 0x02, 0x01, 0x00, 0x01, 0x02 };
+    EnDyExtra* this = THIS;
+    s32 pad;
+    GraphicsContext* localGfxCtx = globalCtx->state.gfxCtx;
+    Vtx* data = (Vtx*)SEGMENTED_TO_VIRTUAL(&D_0601BFB0);
+    s32 i;
+    u8 unk[3];
+    GraphicsContext* gfxCtx = localGfxCtx;
```

Comment:
for the sake of the macro stuff i would just declare the gfxCtx temp and then assign it later like you already have there